app/src/main/java/io/homeassistant/companion/android/settings/vehicle/views/AndroidAutoFavoritesView.kt

```diff
@@ -0,0 +1,132 @@
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.friendlyName
+import io.homeassistant.companion.android.database.server.Server
+import io.homeassistant.companion.android.settings.vehicle.ManageAndroidAutoViewModel
+import io.homeassistant.companion.android.util.compose.FavoriteEntityRow
+import io.homeassistant.companion.android.util.compose.ServerExposedDropdownMenu
+import io.homeassistant.companion.android.util.compose.SingleEntityPicker
+import io.homeassistant.companion.android.vehicle.MainVehicleScreen
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.burnoutcrew.reorderable.ReorderableItem
+import org.burnoutcrew.reorderable.rememberReorderableLazyListState
+import org.burnoutcrew.reorderable.reorderable
+import io.homeassistant.companion.android.common.R as commonR
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun AndroidAutoFavoritesSettings(
+    androidAutoViewModel: ManageAndroidAutoViewModel,
+    serversList: List<Server>,
+    defaultServer: Int
+) {
+    val reorderState = rememberReorderableLazyListState(
+        onMove = { from, to -> androidAutoViewModel.onMove(from, to) },
+        canDragOver = { draggedOver, _ -> androidAutoViewModel.canDragOver(draggedOver) },
+        onDragEnd = { _, _ -> androidAutoViewModel.saveFavorites() }
+    )
+
+    var selectedServer by remember { mutableStateOf(defaultServer) }
+
+    val favoriteEntities = androidAutoViewModel.favoritesList.toList()
+    var validEntities by remember { mutableStateOf<List<Entity<*>>>(emptyList()) }
+    LaunchedEffect(favoriteEntities.size, androidAutoViewModel.sortedEntities.size) {
+        validEntities = withContext(Dispatchers.IO) {
+            androidAutoViewModel.sortedEntities
+                .filter {
+                    !favoriteEntities.contains("$selectedServer-${it.entityId}") &&
+                        (it.entityId.split(".")[0] in MainVehicleScreen.SUPPORTED_DOMAINS)
+                }
+                .toList()
+        }
+    }
+
+    LazyColumn(
+        state = reorderState.listState,
+        contentPadding = PaddingValues(vertical = 16.dp),
+        modifier = Modifier
+            .reorderable(reorderState)
+    ) {
+        item {
+            Text(
+                text = stringResource(commonR.string.aa_set_favorites),
+                fontWeight = FontWeight.Bold,
+                modifier = Modifier.padding(horizontal = 16.dp)
+            )
+        }
+
+        if (serversList.size > 1) {
+            item {
+                ServerExposedDropdownMenu(
+                    servers = serversList,
+                    current = selectedServer,
+                    onSelected = {
+                        androidAutoViewModel.loadEntities(it)
+                        selectedServer = it
+                    },
+                    modifier = Modifier.fillMaxWidth().padding(all = 16.dp)
```

Comment:
When using more than one server there is too much padding between the 2 dropdowns (32dp instead of 16dp). Suggestion for Text above, this dropdown and the single entity picker: on each, set horizontal + bottom padding to fix this -> `.padding(horizontal = 16.dp).padding(bottom = 16.dp)`.

Reply:
done but I noticed that when the server picker is not added the padding is still not good so added a padding for `top` only when we are not adding the server above the entity picker

Reply:
That might be because you didn't update Text in line 74, now we're back to the same issue of [the server dropdown being too close to the text](https://github.com/home-assistant/android/pull/3670#discussion_r1268542404).

Reply:
Ah yes indeed i forgot about that one, fixed :)